### ISSUE TITLE
RuntimeError: cuda runtime error (48) : unrecognized error code at mmdet/ops/roi_align/src/roi_align_kernel.cu:140

### ISSUE THREAD

**Issue by gongtian1234** [open-mmlab/mmdetection#1970]
When testing 'result = inference_detector(model, img)',  I get the following error: 

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/apis/inference.py in 
    inference_detector(model, img)
         84     # forward the model
         85     with torch.no_grad():
    ---> 86         result = model(return_loss=False, rescale=True, **data)
         87     return result
         88 

    /usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py in __call__(self, *input, **kwargs)
        539             result = self._slow_forward(*input, **kwargs)
        540         else:
    --> 541             result = self.forward(*input, **kwargs)
        542         for hook in self._forward_hooks.values():
        543             hook_result = hook(self, input, result)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/core/fp16/decorators.py in new_func(*args, **kwargs)
         47                                 'method of nn.Module')
         48             if not (hasattr(args[0], 'fp16_enabled') and args[0].fp16_enabled):
    ---> 49                 return old_func(*args, **kwargs)
         50             # get the arg spec of the decorated method
         51             args_info = getfullargspec(old_func)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/models/detectors/base.py in forward(self, img, img_meta, return_loss, **kwargs)
        138             return self.forward_train(img, img_meta, **kwargs)
        139         else:
    --> 140             return self.forward_test(img, img_meta, **kwargs)
        141 
        142     def show_result(self, data, result, dataset=None, score_thr=0.3):

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/models/detectors/base.py in forward_test(self, imgs, img_metas, **kwargs)
        121 
        122         if num_augs == 1:
    --> 123             return self.simple_test(imgs[0], img_metas[0], **kwargs)
        124         else:
        125             return self.aug_test(imgs, img_metas, **kwargs)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/models/detectors/two_stage.py in simple_test(self, img, img_meta, proposals, rescale)
        306 
        307         det_bboxes, det_labels = self.simple_test_bboxes(
    --> 308             x, img_meta, proposal_list, self.test_cfg.rcnn, rescale=rescale)
        309         bbox_results = bbox2result(det_bboxes, det_labels,
        310                                    self.bbox_head.num_classes)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/models/detectors/test_mixins.py in simple_test_bboxes(self, x, img_meta, proposals, rcnn_test_cfg, rescale)
        104         rois = bbox2roi(proposals)
        105         roi_feats = self.bbox_roi_extractor(
    --> 106             x[:len(self.bbox_roi_extractor.featmap_strides)], rois)
        107         if self.with_shared_head:
        108             roi_feats = self.shared_head(roi_feats)

    /usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py in __call__(self, *input, **kwargs)
        539             result = self._slow_forward(*input, **kwargs)
        540         else:
    --> 541             result = self.forward(*input, **kwargs)
        542         for hook in self._forward_hooks.values():
        543             hook_result = hook(self, input, result)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/core/fp16/decorators.py in new_func(*args, **kwargs)
        125                                 'method of nn.Module')
        126             if not (hasattr(args[0], 'fp16_enabled') and args[0].fp16_enabled):
    --> 127                 return old_func(*args, **kwargs)
        128             # get the arg spec of the decorated method
        129             args_info = getfullargspec(old_func)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/models/roi_extractors/single_level.py in forward(self, feats, rois, roi_scale_factor)
        103             if inds.any():
        104                 rois_ = rois[inds, :]
    --> 105                 roi_feats_t = self.roi_layers[i](feats[i], rois_)
        106                 roi_feats[inds] = roi_feats_t
        107         return roi_feats

    /usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py in __call__(self, *input, **kwargs)
        539             result = self._slow_forward(*input, **kwargs)
        540         else:
    --> 541             result = self.forward(*input, **kwargs)
        542         for hook in self._forward_hooks.values():
        543             hook_result = hook(self, input, result)

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/ops/roi_align/roi_align.py in forward(self, features, rois)
         78         else:
         79             return roi_align(features, rois, self.out_size, self.spatial_scale,
    ---> 80                              self.sample_num)
         81 
         82     def __repr__(self):

    /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/ops/roi_align/roi_align.py in forward(ctx, features, rois, out_size, spatial_scale, sample_num)
         24         if features.is_cuda:
         25             roi_align_cuda.forward(features, rois, out_h, out_w, spatial_scale,
    ---> 26                                    sample_num, output)
         27         else:
         28             raise NotImplementedError

    RuntimeError: cuda runtime error (48) : unrecognized error code at mmdet/ops/roi_align/src/roi_align_kernel.cu:140

What is the cause of this and how can I solve it?I will be very grateful if you can help me solve this problem

**Comment by hellock** [comment]
CPU inference is currently not implemented.

**Comment by gongtian1234** [comment]
> CPU inference is currently not implemented.

Thanks for your answer, but I use GPU, where do you see it is CPU, thank you for your patience

**Comment by hellock** [comment]
The error log clearly shows the reason.
```
/content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/ops/roi_align/roi_align.py in forward(ctx, features, rois, out_size, spatial_scale, sample_num)
     24         if features.is_cuda:
     25             roi_align_cuda.forward(features, rois, out_h, out_w, spatial_scale,
---> 26                                    sample_num, output)
     27         else:
     28             raise NotImplementedError
```

**Comment by gongtian1234** [comment]
> The error log clearly shows the reason.
> 
> ```
> /content/drive/My Drive/model_train/10_酒瓶检测/mmdetection/mmdet/ops/roi_align/roi_align.py in forward(ctx, features, rois, out_size, spatial_scale, sample_num)
>      24         if features.is_cuda:
>      25             roi_align_cuda.forward(features, rois, out_h, out_w, spatial_scale,
> ---> 26                                    sample_num, output)
>      27         else:
>      28             raise NotImplementedError
> ```

Okay, thanks for your patience, let me think about how to solve this problem

**Comment by gongtian1234** [comment]
1

**Comment by 36572545** [comment]
I don't think it's that reason "CPU NOT implemented", because it's caused by roi_align_cuda.forward(). I am still trapped in this error.

**Comment by AriouatI** [comment]
did you solve the problem @gongtian1234 ?  @36572545 is right  i think it's realted to the cuda version

**Comment by Anuradhabala26** [comment]
I'm getting the same error. Have you solved it? @gongtian1234  @36572545
